server/server_test.go

```diff
@@ -868,6 +868,82 @@ var _ = Describe("Server package test", func() {
 		})
 	})
 
+	Describe("SubstringSearchEnabled", func() {
+		It("should work", func() {
+			testURL("POST", networkPluralURL, adminTokenID, getNetwork("red", "red"), http.StatusCreated)
+			testURL("POST", networkPluralURL, adminTokenID, getNetwork("red1", "red"), http.StatusCreated)
+			testURL("POST", networkPluralURL, adminTokenID, getNetwork("red2", "blue"), http.StatusCreated)
+			testURL("POST", networkPluralURL, adminTokenID, getNetwork("red3", "blue"), http.StatusCreated)
+
+			result := testURL("GET", networkPluralURL, adminTokenID, nil, http.StatusOK)
+			Expect(result).To(HaveKeyWithValue("networks", ConsistOf(
+				util.MatchAsJSON(getNetwork("red", "red")),
+				util.MatchAsJSON(getNetwork("red1", "red")),
+				util.MatchAsJSON(getNetwork("red2", "blue")),
+				util.MatchAsJSON(getNetwork("red3", "blue")))))
+
+			result = testURL("GET", networkPluralURL+"?id=networkred", adminTokenID, nil, http.StatusOK)
```

Comment:
Maybe we can use `~` instead `=` for this feature

Reply:
@stepek..would it be possible to use ~ as a get query condition in GET url..

Kozlowski...what you mentioned is exactly valid and we were also thinking about doing external api for differentiation through a custom action...but these scenario's are few in contrast and disabling the search feature on the schema would be a work around in such cases..but that might not be optimal..in these scenarios we should be blocking sub string match on the granularity level of column instead of the entire table.

Reply:
@giyyanan, you're missing the point: now it's **impossible** in the UI to specify if an exact match or a substring match is required. there are three use cases:
1. a user searches for a resource: we should use substring match/fuzzy search
2. UI sends internal requests: we **must** use exact match (to preserve backward compatibility)
3. an external API user (eg. some custom UI) sends requests: we **must** use exact match (to preserve backward compatibility)

`SubstringSearchEnabled/Disabled`does not solve it. what @stepek suggested (`path/?someField~substring` vs. `path/?someField=exactMatch`) does.

it's not a big deal to implement it, as long as you follow my suggestion with `type Like struct`. entrypoint: [FilterFromQueryParameter](https://github.com/cloudwan/gohan/blob/master/server/resources/resource_management.go#L274)